### DISCUSSION TITLE
How do you develop a plugin without publishing it?

### DISCUSSION THREAD

**Discussion by SamHasler** [oclif/oclif#1477] [Q&A] [answered]
I'm trying refactor functionality out of a command and into a plugin, but I can't see any examples of how to configure a "local" plugin that is not yet published to npm. Is this possible?

**Comment by SamHasler** [comment]
The answer appears to be, you install the plugin using a file uri:
```
./bin/dev.js plugins install file:///C:/PATH/TO/PLUGIN/PROJECT/ROOT
```

I had assumed there would be a way to configure it using the `devplugins` config value in the oclif section of the cli's package.json but I couldn't get it to work with a file uri.

**Comment by mdonnalley** [comment, marked as answer]
I think you're looking for `plugins link`: https://github.com/oclif/plugin-plugins?tab=readme-ov-file#mycli-plugins-link-path

**Reply by SamHasler** [reply]
@mdonnalley Thank you! I looked at the plugins help, don't know how I missed that.
